src/main/java/com/iexec/worker/docker/ComputationService.java

```diff
@@ -81,15 +100,45 @@ public boolean isAppDownloaded(String imageUri) {
         return customDockerClient.isImagePulled(imageUri);
     }
 
-    public boolean runNonTeeComputation(TaskDescription taskDescription,
-                                        WorkerpoolAuthorization workerpoolAuthorization) {
+    /*
+     * non TEE: download secrets && decrypt dataset (TODO: rewritte or remove)
+     *     TEE: download post-compute image && create secure session
+     * 
+     */
+    public void runPreCompute(ComputeMeta computeMeta, TaskDescription taskDescription,
+                WorkerpoolAuthorization workerpoolAuth) {
+        log.info("Running pre-compute [chainTaskId:{}, isTee:{}]", taskDescription.getChainTaskId(),
+                taskDescription.isTeeTask());
+        if (taskDescription.isTeeTask()) {
+            String secureSessionId = runTeePreCompute(taskDescription, workerpoolAuth);
+            computeMeta.setSuccessfulPreCompute(!secureSessionId.isEmpty());
```

Comment:
Detail - Probably could save 1 call computeMeta.setSuccessfulPreCompute(..) in the method. At least we could have same computeMeta update way in runPreCompute & runPostCompute.

Reply:
Updated thanks!